File: post_3/serving/fastapi-onnx/app/routes/root.py
from fastapi import Request


def read_root(request: Request):
    model = request.app.state.model
    return {
        "message": "Adult Income Prediction API",
        "status": "Model loaded" if model.is_loaded() else "Model not loaded",
        "endpoints": {
            "POST /predict": "Make a prediction",
            "POST /predict/batch": "Make batch predictions",
            "GET /health": "Check API health",
        },
    }

File: post_3/serving/axum-onnx/src/routes/root.rs
use axum::Json;
use crate::types::{RootResponse, EndpointsInfo};

pub async fn root() -> Json<RootResponse> {
    Json(RootResponse {
        message: "Adult Income Prediction API (ONNX Runtime)".to_string(),
        status: "Model loaded".to_string(),
        endpoints: EndpointsInfo {
            post_predict: "Make a prediction".to_string(),
            get_health: "Check API health".to_string(),
        },
    })
}
